### PR TITLE
Adding Asset Mapper support + new StimulusBundle

### PR DESCRIPTION
**Pull request by weaverryan**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Tickets       | None
| License       | MIT

Adds:

* A) Integration with new `symfony.importmap` key introduced to Flex in https://github.com/symfony/flex/pull/975
* B) An "asset mapper" path in each bundle so that you can refer to the controller/files in that bundle.
* C) Introduces new StimulusBundle (thanks to @jmsche for helping port from WebpackEncoreBundle). This includes the Twig `stimulus_*` functions + a revamped "helper" system for generating those attributes.
* D) The new StimulusBundle also contains integration for AssetMapper to load custom controllers and `controllers.json`.

Cheers!

### CODE REVIEW

**Comment by weaverryan** [comment]
This is ready for review - test failures are expected - the "low deps" aren't quite smart enough in the CI to know how to handle dependencies against packages that were added/updated inside the same PR. The "high deps" handle it correctly.

File: .github/build-packages.php [hunk 1]
```diff
@@ -1,21 +1,53 @@
 <?php
 
-$dir = __DIR__.'/../src/LiveComponent';
-$flags = JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE;
+/**
+ * Upates the composer.json files to use the local version of the Symfony UX packages.
```
Comment (nicolas-grekas) [inline .github/build-packages.php]:
```suggestion
 * Updates the composer.json files to use the local version of the Symfony UX packages.
```

File: .github/build-packages.php [hunk 2]
```diff
@@ -1,21 +1,53 @@
...
+        || isset($packageData['require-dev']['symfony/stimulus-bundle'])
+    ) {
+        $repositories[] = [
+            'type' => 'path',
+            'url' => '../StimulusBundle',
+        ];
+        $key = isset($packageData['require']['symfony/stimulus-bundle']) ? 'require' : 'require-dev';
+        $packageData[$key]['symfony/stimulus-bundle'] = '@dev';
+    }
+    if (count($repositories) > 0) {
```
Comment (nicolas-grekas) [inline .github/build-packages.php]:
```suggestion
    if ($repositories) {
```

File: src/Autocomplete/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## 2.9.0
+
+-   Support for symfony/asset-mapper was added.
```
Comment (nicolas-grekas) [inline src/Autocomplete/CHANGELOG.md]:
```suggestion
-   Add support for symfony/asset-mapper
```

File: src/Chartjs/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,13 @@
 # CHANGELOG
 
+## 2.9.0
+
+-   Support for symfony/asset-mapper was added.
```
Comment (nicolas-grekas) [inline src/Chartjs/CHANGELOG.md]:
```suggestion
-   Add support for symfony/asset-mapper
```

File: src/Chartjs/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,13 @@
...
+
+-   symfony/stimulus-bundle was added as a dependency.
```
Comment (nicolas-grekas) [inline src/Chartjs/CHANGELOG.md]:
```suggestion
-   Add dependency on symfony/stimulus-bundle
```

File: src/Chartjs/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,13 @@
...
+
+-   Minimum required PHP version is now 8.0.
```
Comment (nicolas-grekas) [inline src/Chartjs/CHANGELOG.md]:
```suggestion
-   Minimum required PHP version is now 8.1
```

File: src/Chartjs/composer.json [hunk 7]
```diff
@@ -28,21 +28,20 @@
         }
     },
     "require": {
-        "php": ">=7.2.5",
+        "php": ">=8.0",
```
Comment (nicolas-grekas) [inline src/Chartjs/composer.json]:
```suggestion
        "php": ">=8.1",
```

to be consistent with other recent Symfony packages

File: src/Chartjs/composer.json [hunk 8]
```diff
@@ -28,21 +28,20 @@
...
         "symfony/config": "^4.4.17|^5.0|^6.0",
         "symfony/dependency-injection": "^4.4.17|^5.0|^6.0",
-        "symfony/http-kernel": "^4.4.17|^5.0|^6.0"
+        "symfony/http-kernel": "^4.4.17|^5.0|^6.0",
```
Comment (nicolas-grekas) [inline src/Chartjs/composer.json]: let's bump everything to 5.4 for sanity :)

File: src/Chartjs/src/Twig/ChartExtension.php [hunk 9]
```diff
@@ -27,19 +26,27 @@ class ChartExtension extends AbstractExtension
 {
     private $stimulus;
 
-    public function __construct(StimulusTwigExtension $stimulus)
+    /**
+     * @param $stimulus StimulusHelper|StimulusTwigExtension
```
Comment (nicolas-grekas) [inline src/Chartjs/src/Twig/ChartExtension.php]:
```suggestion
     * @param $stimulus StimulusHelper
```

File: src/Chartjs/src/Twig/ChartExtension.php [hunk 10]
```diff
@@ -27,19 +26,27 @@ class ChartExtension extends AbstractExtension
...
+     */
+    public function __construct($stimulus)
```
Comment (nicolas-grekas) [inline src/Chartjs/src/Twig/ChartExtension.php]:
```suggestion
    public function __construct(StimulusHelper|StimulusTwigExtension $stimulus)
```

File: src/Cropperjs/CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## 2.9.0
+
+-   Support for symfony/asset-mapper was added.
```
Comment (nicolas-grekas) [inline src/Cropperjs/CHANGELOG.md]:
```suggestion
-   Add support for symfony/asset-mapper
```

File: src/Dropzone/CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## 2.9.0
+
+-   Support for symfony/asset-mapper was added.
```
Comment (nicolas-grekas) [inline src/Dropzone/CHANGELOG.md]:
```suggestion
-   Add support for symfony/asset-mapper
```

File: src/LazyImage/CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## 2.9.0
+
+-   Support for symfony/asset-mapper was added.
```
Comment (nicolas-grekas) [inline src/LazyImage/CHANGELOG.md]:
```suggestion
-   Add support for symfony/asset-mapper
```

File: src/LiveComponent/CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## 2.9.0
+
+-   Support for symfony/asset-mapper was added.
```
Comment (nicolas-grekas) [inline src/LiveComponent/CHANGELOG.md]:
```suggestion
-   Add support for symfony/asset-mapper
```

File: src/Notify/CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,12 @@
 # CHANGELOG
 
+## 2.9.0
+
+-   Support for symfony/asset-mapper was added.
```
Comment (nicolas-grekas) [inline src/Notify/CHANGELOG.md]:
```suggestion
-   Add support for symfony/asset-mapper
```

File: src/Notify/CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,12 @@
...
+
+-   The `symfony/webpack-encore-bundle` dependency was replaced by
```
Comment (nicolas-grekas) [inline src/Notify/CHANGELOG.md]:
```suggestion
-   Replace `symfony/webpack-encore-bundle` by `symfony/stimulus-bundle` in dependencies
```

File: src/Notify/src/Twig/NotifyRuntime.php [hunk 17]
```diff
@@ -12,23 +12,33 @@
...
 
 /**
  * @author Mathias Arlaud <mathias.arlaud@gmail.com>
  */
 final class NotifyRuntime implements RuntimeExtensionInterface
 {
+    private StimulusHelper $stimulusHelper;
+
+    /**
+     * @param $stimulus StimulusHelper|StimulusTwigExtension
```
Comment (nicolas-grekas) [inline src/Notify/src/Twig/NotifyRuntime.php]:
```suggestion
     * @param $stimulus StimulusHelper
```

File: src/Notify/src/Twig/NotifyRuntime.php [hunk 18]
```diff
@@ -12,23 +12,33 @@
...
+     */
     public function __construct(
         private HubInterface $hub,
-        private StimulusTwigExtension $stimulusTwigExtension,
+        StimulusHelper $stimulus,
```
Comment (nicolas-grekas) [inline src/Notify/src/Twig/NotifyRuntime.php]:
```suggestion
        StimulusHelper|StimulusTwigExtension $stimulus,
```

File: src/React/composer.json [hunk 19]
```diff
@@ -28,7 +28,7 @@
         }
     },
     "require": {
-        "symfony/webpack-encore-bundle": "^1.11"
+        "symfony/stimulus-bundle": "^2.9"
     },
     "require-dev": {
         "symfony/framework-bundle": "^4.4|^5.0|^6.0",
```
Comment (nicolas-grekas) [inline src/React/composer.json]: let's bump everything to 5.4 in the same ?

File: src/React/CHANGELOG.md [hunk 20]
```diff
@@ -1,5 +1,12 @@
 # CHANGELOG
 
+## 2.9.0
+
+-   Support for symfony/asset-mapper was added.
+
+-   The `symfony/webpack-encore-bundle` dependency was replaced by
+    `symfony/stimulus-bundle`.
```
Comment (nicolas-grekas) [inline src/React/CHANGELOG.md]: same suggestion as above (and same in every similar file if you accept it :) )

File: src/React/src/Twig/ReactComponentExtension.php [hunk 21]
```diff
@@ -23,27 +23,38 @@
  */
 class ReactComponentExtension extends AbstractExtension
 {
-    private $stimulusExtension;
+    private $stimulusHelper;
 
-    public function __construct(StimulusTwigExtension $stimulusExtension)
+    /**
+     * @param $stimulus StimulusHelper|StimulusTwigExtension
```
Comment (nicolas-grekas) [inline src/React/src/Twig/ReactComponentExtension.php]:
```suggestion
     * @param $stimulus StimulusHelper
```

File: src/React/src/Twig/ReactComponentExtension.php [hunk 22]
```diff
@@ -23,27 +23,38 @@
...
+     */
+    public function __construct($stimulus)
```
Comment (nicolas-grekas) [inline src/React/src/Twig/ReactComponentExtension.php]:
```suggestion
    public function __construct(StimulusHelper|StimulusTwigExtension $stimulus)
```

File: src/StimulusBundle/CHANGELOG.md [hunk 23]
```diff
@@ -0,0 +1,5 @@
+# CHANGELOG
+
+## 2.9.0
+
+-   The bundle was first introduced.
```
Comment (nicolas-grekas) [inline src/StimulusBundle/CHANGELOG.md]:
```suggestion
-   Introduce the bundle
```

File: src/StimulusBundle/README.md [hunk 24]
```diff
@@ -0,0 +1,11 @@
+# StimulusBundle: Symfony integration with Stimulus!
+
+This bundle adds integration between Symfony, Stimulus and Symfony UX:
+
+-   A) Twig `stimulus_*` functions & filters to add Stimulus controllers, actions & targets in your templates.
```
Comment (nicolas-grekas) [inline src/StimulusBundle/README.md]:
```suggestion
-   A) Twig `stimulus_*` functions & filters to add Stimulus controllers, actions & targets in your templates;
```

File: src/StimulusBundle/README.md [hunk 25]
```diff
@@ -0,0 +1,11 @@
...
+-   B) Integration with Symfony UX & AssetMapper
```
Comment (nicolas-grekas) [inline src/StimulusBundle/README.md]:
```suggestion
-   B) Integration with Symfony UX & AssetMapper;
```

File: src/StimulusBundle/composer.json [hunk 26]
```diff
@@ -0,0 +1,41 @@
...
+    "license": "MIT",
+    "type": "symfony-bundle",
+    "authors": [
+        {
+            "name": "Symfony Community",
+            "homepage": "https://symfony.com/contributors"
+        }
+    ],
+    "require": {
+        "php": ">=8.0",
```
Comment (nicolas-grekas) [inline src/StimulusBundle/composer.json]:
```suggestion
        "php": ">=8.1",
```

File: src/StimulusBundle/config/services.php [hunk 27]
```diff
@@ -0,0 +1,48 @@
...
+use Twig\Environment;
+
+use function Symfony\Component\DependencyInjection\Loader\Configurator\abstract_arg;
+use function Symfony\Component\DependencyInjection\Loader\Configurator\param;
+use function Symfony\Component\DependencyInjection\Loader\Configurator\service;
+
+return static function (ContainerConfigurator $container): void {
+    $container->services()
+        ->set('stimulus.helper', StimulusHelper::class)
+            ->arg('$twig', service(Environment::class)->nullOnInvalid())
```
Comment (nicolas-grekas) [inline src/StimulusBundle/config/services.php]: better not force using reflection when we can avoid it:
```suggestion
            ->args([service(Environment::class)->nullOnInvalid()])
```

File: src/StimulusBundle/doc/index.rst [hunk 28]
```diff
@@ -0,0 +1,313 @@
+StimulusBundle: Symfony integration with Stimulus
+=================================================
+
+This bundle adds integration between Symfony, `Stimulus`_ and Symfony UX:
+
+A) Twig `stimulus_*` functions & filters to add Stimulus controllers, actions & targets in your templates.
```
Comment (nicolas-grekas) [inline src/StimulusBundle/doc/index.rst]:
```suggestion
A) Twig `stimulus_*` functions & filters to add Stimulus controllers, actions & targets in your templates;
```

File: src/StimulusBundle/doc/index.rst [hunk 29]
```diff
@@ -0,0 +1,313 @@
...
+B) Integration with Symfony UX & AssetMapper
```
Comment (nicolas-grekas) [inline src/StimulusBundle/doc/index.rst]:
```suggestion
B) Integration with Symfony UX & AssetMapper;
```

File: src/StimulusBundle/doc/index.rst [hunk 30]
```diff
@@ -0,0 +1,313 @@
...
+C) A helper service to build the Stimulus data attributes and use them in your services.
+
+Installation
+------------
+
+First, if you don't have one yet, choose and install an asset handling system:
```
Comment (nicolas-grekas) [inline src/StimulusBundle/doc/index.rst]:
```suggestion
First, if you don't have one yet, choose and install an asset handling system;
```

**Pull request merged**